File: app/src/main/java/com/github/panda_io/micro_panda/builder/llvm/ir/constant/Null.java
package com.github.panda_io.micro_panda.builder.llvm.ir.constant;

import com.github.panda_io.micro_panda.builder.llvm.ir.type.Type;

public class Null extends Constant {
    Type type;

    public Null(Type type) {
        this.type = type;
    }

    public String string() {
        return String.format("%s null", this.type.string());
    }

    public Type getType() {
        return this.type;
    }

    public String identifier() {
        return "null";
    }
}

File: app/src/main/java/com/github/panda_io/micro_panda/builder/llvm/ir/constant/null.go
package constant

import (
	"fmt"

	"github.com/panda-io/micro-panda/target/llvm/ir/ir"
	"github.com/panda-io/micro-panda/target/llvm/ir/ir_types"
)

type Null struct {
	Typ *ir_types.PointerType
}

func NewNull(typ *ir_types.PointerType) *Null {
	return &Null{Typ: typ}
}

func (c *Null) String() string {
	return fmt.Sprintf("%s %s", c.Type().String(), c.Ident())
}

func (c *Null) Type() ir.Type {
	return c.Typ
}

func (*Null) Ident() string {
	return "null"
}
